### PR TITLE
Simplified workflow form runtime options

### LINKED ISSUE #12595: Unexpanded workflow forms don't allow export to new history
Screen Shot 2021-09-29 at 10 30 59 AM

I don't know if it's intentional, but workflows pages that have not been expanded to show every step do not give the option for the user to export results to another history.

To reproduce: Click to run a workflow that does not expand by default

Expected behavior: "history options" boolean input at the top of the workflow, which allows export to a new user-named history

Actual behavior: No option available

### PR DESCRIPTION
**Pull request by jmchilton**
Includes all my various workflow testing improvements from the deferred data branch also.

Fixes https://github.com/galaxyproject/galaxy/issues/12595

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/components/Workflow/Run/WorkflowRunFormSimple.vue [hunk 1]
```diff
@@ -37,9 +67,12 @@ export default {
         },
     },
     data() {
+        const newHistory = this.targetHistory == "new" || this.targetHistory == "prefer_new";
         return {
             formData: {},
             inputTypes: {},
+            sendToNewHistory: newHistory,
+            useCachedJobs: this.useJobCache, // TODO:
```
Comment (bgruening) [inline client/src/components/Workflow/Run/WorkflowRunFormSimple.vue:75]: Is a to-do missing here? Or can this be removed?

**Review by davelopez — APPROVED** [review]
Looks good and works! Thank you @jmchilton!

Just needs a rebase to resolve the conflicts :)

File: client/src/components/Workflow/Run/WorkflowRunFormSimple.vue (hunk 1 above)
Comment (jmchilton) [reply client/src/components/Workflow/Run/WorkflowRunFormSimple.vue:75]: I'd like to add more configurability here but I think the current logic works. The value in galaxy.yml will serve as the default value but if user preferences are set they will be used to make this configurable.

Comment (mvdbeek) [reply client/src/components/Workflow/Run/WorkflowRunFormSimple.vue:75]: sendToNewHistory and useCachedJobs should be computed properties, I think

Comment (jmchilton) [reply client/src/components/Workflow/Run/WorkflowRunFormSimple.vue:75]: There is a v-model on each of them - I thought that should target data properties?

Comment (mvdbeek) [reply client/src/components/Workflow/Run/WorkflowRunFormSimple.vue:75]: Hmm, I don't know what would be the right thing here in combination with v-model. The way you compute newHistory within `data()` looks odd to me and I wouldn't do it if there was no v-model, which is why I thought this isn't quite right. I guess you wouldn't want newHistory to be reactive, so maybe this is the right thing to do ?

**Review by bgruening — APPROVED** [review]

**Pull request merged**